src/component.cpp

```diff
@@ -137,8 +137,16 @@ bool Component::ComponentImpl::performTestWithHistory(History &history, const Co
     for (const auto &u : units) {
         if ((type == TestType::RESOLVED) && !u->isResolved()) {
             return false;
-        } else if ((type == TestType::DEFINED) && !u->isDefined()) {
-            return false;
+        } else if (type == TestType::DEFINED) {
+            if (!u->isDefined()) {
+                return false;
+            }
+            if (model == nullptr) {
+                return false;
+            }
+            if (!model->hasUnits(u)) {
+                return false;
+            }
```

Comment:
Why not simply have:
```c++
if ((model == nullptr) || !model->hasUnits(u) || !u->isDefined()) {
    return false;
}
```

Reply:
[be57464](https://github.com/cellml/libcellml/pull/1167/commits/be57464f44dd1c778b5bc4d3b1b9d8554af9a5ca)